Add writing of metadata into sim_telarray configuration files.

GernotMaier opened the pull request:
Add functionality to write metadata lines into the sim_telarray configuration files:

- uses new `meta_parameter` field in the model_parameter data dictionaries to decide if `metapara` should be set for a model parameters
- adds `global` for site and `telescope` for telescope configuration files.
- required to move the functionality of `model_parameters.get_simtel_parameters` into `simtel_config_writer`
- (I've removed at the two functions nowhere used in `model_parameter`, after stumbling over them)

Closes #1131 
Related to merge request [31](https://gitlab.cta-observatory.org/cta-science/simulations/simulation-model/simulation-models/-/merge_requests/31), [28](https://gitlab.cta-observatory.org/cta-science/simulations/simulation-model/simulation-models/-/merge_requests/28) and issue [4](https://gitlab.cta-observatory.org/cta-science/simulations/simulation-model/simulation-models/-/issues/15)

orelgueta commented on src/simtools/model/model_parameter.py:
```diff
@@ -421,29 +373,24 @@ def change_parameter(self, par_name, value):
-            value = gen.convert_string_to_list(value)
+        value = gen.convert_string_to_list(value) if isinstance(value, str) else value
 
+        param_type = self.get_parameter_type(par_name)
```
Not crazy about `param_type` and `par_name` being inconsistent abbreviations.

GernotMaier replied:
Fixed.

orelgueta reviewed (commented):
Do I understand it correctly that there are no unit tests that check that the functions write out the meta parameters correctly (global, set, add, etc.)?

GernotMaier commented:
> Do I understand it correctly that there are no unit tests that check that the functions write out the meta parameters correctly (global, set, add, etc.)?

Yes, you understood correctly. Fixed now - I've added an improved version of the unit tests.

orelgueta commented on tests/unit_tests/simtel/test_simtel_config_writer.py:
```diff
@@ -132,3 +134,41 @@ def test_convert_model_parameters_to_simtel_format(
+    assert f"optics_config_name = {simtel_config_writer._telescope_model_name}" in _tel
+
+    _site = simtel_config_writer._get_simtel_metadata("site", None)
+    assert len(_site) == 8
```
I suggest to remove this assert, it's not very useful and will just be annoying if in the future we add/remove a parameter.

GernotMaier replied:
Removed.

orelgueta commented on tests/unit_tests/simtel/test_simtel_config_writer.py:
```diff
@@ -132,3 +134,41 @@ def test_convert_model_parameters_to_simtel_format(
+        side_effect=mock_get_name,
+    ):
+        tel_meta = simtel_config_writer._get_simtel_metadata("telescope", model_parameters)
+        assert "metaparam telescope add test_param" in tel_meta
+        assert "metaparam telescope set test_param_meta=42" in tel_meta
```
As mentioned, this and the code made me worry that we always write `add` and `set` in the cfg files, but it doesn't look like it. So, no need to worry I think.

orelgueta reviewed (approved):
Approving already since the cfg files look OK and the comment I have is very minor. The fact the mocking unit test confuses me shouldn't be an issue.

GernotMaier commented:
@orelgueta - thanks for the review.

I've updated the unit tests to make the return values for adding and setting clearer, see the test_get_simtel_metadata_with_model_parameters function.
